test/integration/types_test.rb

```diff
@@ -268,6 +268,74 @@ def test_escaped_data
     end
   end
 
+  def test_copy_data_in_fixed_size_binary_column
+    # test for the BINARY columns needing 0-byte padding
+    #
+    # For details, see https://github.com/Shopify/ghostferry/pull/159
+
+    [source_db, target_db].each do |db|
+      db.query("CREATE DATABASE IF NOT EXISTS #{DEFAULT_DB}")
+      db.query("CREATE TABLE IF NOT EXISTS #{DEFAULT_FULL_TABLE_NAME} (id bigint(20) not null auto_increment, data BINARY(4), primary key(id))")
```

Comment:
> Can we add a test for a case when the length is 0 (ex: create table a (d binary);) 

sure

>  and for the decimal type above?

Just to avoid an unnecessary round-trip: what do you mean by that?

Reply:
In the `decimal.Decimal` case, the function is called with a length of `0` - `appendEscapedString(buffer, v.String(), 0)`. I meant a test case of length `0` would cover both the `decimal.Decimal` case as well as if the length of the column was `0` (ie: `create table a (d binary);`).

Reply:
got it. I thought we already had a test-case for decimals, but I'll check.